src/sched_rebuild.c

```diff
@@ -38,9 +38,35 @@ struct sched_class *mod_class[] = {
 	&shadow_idle_sched_class,
 };
 
+DEFINE_PER_CPU(struct list_head, dying_task_list);
+
 #define NR_SCHED_CLASS 5
 struct sched_class bak_class[NR_SCHED_CLASS];
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 3, 0)
+static void put_prev_task_fake(struct rq *rq, struct task_struct *prev)
+{
+}
+
+static const struct sched_class fake_sched_class = {
+	.put_prev_task = put_prev_task_fake,
+};
+
+static struct task_struct fake_task = {
+	/*
+	 * Avoid pull_{rt,dl}_task()
+	 */
+	.prio = MAX_PRIO + 1,
+	.sched_class = &fake_sched_class,
+};
```

Comment:
Can we just use the vmlinux's fake_task?
```c
extern struct task_struct __orig_fake_task;
```

Reply:
Yes we can. But I worry about it maybe optimized.

Reply:
If `fake_task` is optimized, then `migrate_tasks` in the module won't find `fake_task` either, and the whole scheduler.ko won't be loaded.

It won't help anything, if we merely solve this in main.c.

Also, gcc don't usually optimize global variables. I prefer keep this code snippet as simple as possible.